cmd/process-agent/collector.go

```diff
@@ -273,7 +273,7 @@ func errResponse(format string, a ...interface{}) postResponse {
 	return postResponse{err: fmt.Errorf(format, a...)}
 }
 
-func (l *Collector) postToAPI(endpoint config.APIEndpoint, checkPath string, body []byte, responses chan postResponse) {
+func (l *Collector) postToAPI(endpoint config.APIEndpoint, checkPath string, body []byte, responses chan postResponse, mb model.MessageBody) {
```

Comment:
nit: can we pass in a `hasContainers bool` here instead of the whole message body and move the call to hasContainers to `postMessage`? That way we're not passing in the message in both encoded and unencoded form.